test/unit/testplan/web_ui/test_web_app.py

```diff
@@ -0,0 +1,202 @@
+
+
+class TestStaticEndpoints(object):
+    """
+    Test the endpoints returning files from the STATIC_PATH directory.
+    """
+
+    def setup_method(self, _):
+        """Create fake static files and a test client."""
+        self.static_dir = tempfile.mkdtemp()
+        _create_tmp_static_files(self.static_dir)
+        tp_web_app.config['STATIC_PATH'] = self.static_dir
+        tp_web_app.config['TESTING'] = True
+        self.client = tp_web_app.test_client()
+
+    def teardown_method(self, _):
+        """Remove fake static files."""
+        shutil.rmtree(self.static_dir)
+
+    def test_testplan_index(self):
+        """Does /testplan/<uid> return the correct index.html page."""
+        # Use correct UID, expect 200 response.
+        response = self.client.get('/testplan/{}'.format(STATIC_REPORTS['testing']['uid']))
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['testing']['contents']) in str(response.data)
+
+        # Use incorrect UID, still expect 200 response.
+        response = self.client.get('/testplan/123')
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['testing']['contents']) in str(response.data)
+
+    def test_testplan_static(self):
+        """Does /testplan/static/static.file respond with the static.file file."""
+        # Use correct UID, expect 200 response
+        response = self.client.get('/testplan/static/static.file'.format(STATIC_REPORTS['testing']['uid']))
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['testing']['contents']) in str(response.data)
+
+        # This endpoint should return the index.html page no matter what UID is sent
+        response = self.client.get('/testplan/static/123')
+        assert response.status_code == 404
+        assert str(STATIC_REPORTS['testing']['contents']) not in str(response.data)
+
+    def test_monitor_index(self):
+        """Does /monitor/<uid> return the correct index.html page."""
+        # Use correct UID, expect 200 response.
+        response = self.client.get('/monitor/{}'.format(STATIC_REPORTS['monitor']['uid']))
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['monitor']['contents']) in str(response.data)
+
+        # Use incorrect UID, still expect 200 response.
+        response = self.client.get('/monitor/123')
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['monitor']['contents']) in str(response.data)
+
+    def test_monitor_static(self):
+        """Does /testplan/static/static.file respond with the static.file file."""
+        # Use correct UID, expect 200 response
+        response = self.client.get('/monitor/static/static.file'.format(STATIC_REPORTS['monitor']['uid']))
+        assert response.status_code == 200
+        assert str(STATIC_REPORTS['monitor']['contents']) in str(response.data)
+
+        # This endpoint should return the index.html page no matter what UID is sent
```

Comment:
again, I don't think this comment matches the code

Reply:
Thanks I'll change it